### PR TITLE
deprecate `flipdim` to `reverse` with a `dims` keyword argument

### PR DESCRIPTION
**Pull request by JeffBezanson**
I think this is another case that fits well with the `dims` keyword argument change. The docstring for `flipdim` describes the operation as "reverse", often a telltale sign.

### CODE REVIEW

**Review by mbauman — APPROVED** [review]

**Comment by JeffBezanson** [comment]
Triage accepts.

File: base/abstractarraymath.jl [hunk 1]
```diff
@@ -137,13 +137,13 @@ julia> b = [1 2; 3 4]
...
 
-julia> flipdim(b,2)
+julia> reverse(b, dims=2)
 2×2 Array{Int64,2}:
  2  1
  4  3
 ```
 """
-function flipdim(A::AbstractArray, d::Integer)
+function reverse(A::AbstractArray; dims::Integer)
```
Comment (ararslan) [inline base/abstractarraymath.jl]: Doesn't this overwrite the `reverse` method for arrays with one with a required keyword argument?

Comment (mbauman) [reply base/abstractarraymath.jl]: That one is only defined for `AbstractVector`

Comment (ararslan) [reply base/abstractarraymath.jl]: Ah, I see, thanks.

File: base/abstractarraymath.jl [hunk 2]
```diff
@@ -137,13 +137,13 @@ julia> b = [1 2; 3 4]
...
     nd = ndims(A)
     1 ≤ d ≤ nd || throw(ArgumentError("dimension $d is not 1 ≤ $d ≤ $nd"))
```
Comment (ararslan) [inline base/abstractarraymath.jl]: This should be updated to reflect the change from the argument being called `d` to being called `dims`.

Comment (JeffBezanson) [reply base/abstractarraymath.jl]: ooops

**Comment by stevengj** [comment]
Should the old `reverse(::AbstractVector)` functions now take a `dims=1` keyword argument?

Otherwise it seems like it may be a problem for generic code calling `reverse(A, dims=1)` where `A` is sometimes a matrix and sometimes a vector, for example, since in the latter case it dispatches to a method that doesn't take keywords?

**Comment by JeffBezanson** [comment]
Yes, you're right.

**Comment by JeffBezanson** [comment]
Ah, actually it happens to work currently due to #9498 --- passing a keyword argument causes us to favor a matching method that accepts them.

**Comment by stevengj** [comment]
Would be good to at least have a test for `reverse([1,2,3], dim=1) == [3,2,1]` if we don't have one already.

**Comment by stevengj** [comment]
Would also be good to have an in-place `flipdim!`, i.e. a `reverse!` method that accepts the `dim` keyword, but that is a separate issue.

**Pull request merged**